DSpace/dspace-angular
# Fix for #1602 Insufficient collection information when submitting an article

### Description

Fixes #1602 

Insufficient collection information when submitting an article.
### Instructions for Reviewers
Sometimes, the collections returned by the collection browser are indistinguishable because their names are identical to other collections and their parent communities.
For example, suppose you have two communities named "Institution A" and "Institution B," each with a subcommunity named "Reports." Both subcommunities have a collection called "Research Reports." If I want to submit an item to "Institution B > Reports > Research Reports," it's impossible to differentiate between the "Reports" subcommunity in "Institution A" and "Institution B" when searching for the "Research Reports" collection.
**Expected Behavior**
Instead of displaying only the name of the parent community, the interface should display the full hierarchy, similar to how it's done in XMLUI.
### List of Changes in this PR:
•	Added a method getHierarchicalName in sidebar-search-list-element.component.ts that returns the hierarchical name. Now, parentTitle calls getHierarchicalName instead of getParentTitle.
•	Added a method createInstanceFromDSpaceObject responsible for creating an instance of DSpaceObject.
•	Removed the maxlines argument from sidebar-search-list-element.component.html.
•	Added some CSS to the content class in truncatable-part.component.scss.
### Steps to Test and Reproduce the Behavior:
1.	Create two top-level communities named "A" and "B."
2.	Create a subcommunity named "Reports" under each top-level community.
3.	Create a collection named "Research Reports" within each "Reports" subcommunity.
4.	Try to submit an item to "B > Reports > Research Reports" and ensure that the full hierarchical name is displayed.
